### PR TITLE
setup identity

### PR DESCRIPTION
**Pull request by GrinchArt**
dev
## Summary of issue

ToDo

## Summary of change

ToDo

## Comments

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=50%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.WebApi/Program.cs [hunk 1]
```diff
@@ -18,6 +22,14 @@
 builder.Services.ConfigureInstagram(builder);
 builder.Services.ConfigureSerilog(builder);
 
+builder.Services.AddDbContext<StreetcodeDbContext>(options =>
+    options.UseSqlServer(builder.Configuration.GetConnectionString("DefaultConnection")));
+
+builder.Services.AddIdentity<User, IdentityRole<int>>()
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.WebApi/Program.cs:28]: Extra services should not be registered directly in `Program.cs`. Move this into `Streetcode.WebApi.Extensions.ServiceCollectionExtesions`. Either in `AddApplicationServices` or custom helper method similar to `AddModelValidationService`.

**Review by jimmy2jk — CHANGES_REQUESTED** [review]
Impossible to apply migrations. Neither on project startup, running `dotnet ef database update` in console nor deleting and recreating database from zero work correctly. Entire application crashes.

File: Streetcode/Streetcode.DAL/Entities/Users/User.cs [hunk 2]
```diff
@@ -25,6 +26,12 @@ public class User
         [Required]
         [MaxLength(20)]
         public string Password { get; set; } = null!;
```
Comment (jimmy2jk) [inline Streetcode/Streetcode.DAL/Entities/Users/User.cs:28]: Try to change to `PasswordHash` from `IdentityUser`

**Comment by MishaHMK** [comment]
> Impossible to apply migrations. Neither on project startup, running `dotnet ef database update` in the console nor deleting and recreating the database from zero work correctly—the entire application crashes.

Same 

1. You should override properties inherited from the class. Read warnings for props in User and DbContext
2. Register services in 'WebApi/Extensions/ServiceCollectionExtensios', not in 'Program.cs'
3. DbContext is already registered in this application
4. Tables from your migration do not create, after 'Update-Database' command, therefore app doesn't run

**Review by MishaHMK — CHANGES_REQUESTED** [review]
1. You should explicitly write properties inherited from the Identity class. Read warnings for props in User and DbContext
2. Register services in 'WebApi/Extensions/ServiceCollectionExtensios', not in 'Program.cs'
3. DbContext is already registered in this application
4. Tables from your migration do not create, after 'Update-Database' command, therefore app doesn't run

**Comment by jimmy2jk** [comment]
Potential issue in absence of `UserName` attribute in User (renamed to AspNetUsers) table. Seems like migration doesn't finish succesfully, changes are not applied at all. And while seeding DB app tries to execute query to non-existing tables

**Review by jimmy2jk — APPROVED** [review]

**Review by MishaHMK — APPROVED** [review]

**Pull request merged**